geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/executor/sortedset/ZAddExecutor.java

```diff
@@ -77,10 +77,14 @@ private int findAndValidateZAddOptions(Command command, Iterator<byte[]> command
           executorState.chFound = true;
           optionsFoundCount++;
           break;
+        case "-inf":
+        case "inf":
+        case "+inf":
+          scoreFound = true;
+          break;
         default:
-          try {
-            Double.valueOf(subCommandString);
-          } catch (NumberFormatException nfe) {
+          // Use regex to validate score is a valid floating point number
+          if (!subCommandString.matches("^[-+]?[0-9]*\\.?[0-9]+([eE][-+]?[0-9]+)?$")) {
```

Comment:
Well, the intent is to avoid the whole conversion, and just do the regex. Looking at, e.g. OpenJDK's implementation of parseDouble, though, they pretty much do a regex anyway. There might possibly be one extra stack frame, but it shouldn't matter particularly. So I swapped it for just Double.parseDouble().